summary.Rmd

```diff
@@ -0,0 +1,40 @@
+Summary
+=======
+
+1. Use available tools
+    a. Use ready-made images (e.g. from Docker Hub)
+    b. Dockerfile generators (e.g. repo2docker)
+    c. Project templates (e.g. [cookie cutter](https://github.com/cookiecutter/cookiecutter))
+    d. Linters
+2. Use versioned images
+3. Format for clarity
+    a. Use indentation, line breaks and unabbreviated arguments
+    b. Clarity > brevity
+    c. Group related commands
+4. Document within the Dockerfile
+    a. Add comments explaining the steps
+    b. Use labels for metadata
+    c. Include build and run commands
+6. Specify software versions
+    a. Pin system library versions
+    b. Explictly list package versions to install
+    c. Comment why each is needed
+9. Use version control
+    a. Use Continuous Integration and automated testing
+    b. ~~Use one Dockerfile per project~~
+    c. ~~Use project templates (e.g. cookie cutter)~~
+7. Mount user data
+    a. Use bind mount for accessibility
+    b. Keep large or sensitive datasets out with `.dockerignore`
+    c. Document expected mounts for external resources
+5. Order the instructions
+    a. Least --> most changable for caching and fast builds
+    b. Regularly use `--no-cache` to catch problems
+8. Make it one-click runnable
+    a. Set default behaviour with `CMD` and `ENTRYPOINT`
+10. Regularly use and rebuild containers
+    a. Use the container throughout each project for confidence in your workflow
+    b. Rebuild every 1-2 weeks with `--no-cache` to catch problems
+    c. Provide a `Makefile` or script for convenience
+    d. Ask a colleague to try it
+    e. Push the image to a public registry at publication time
```

Comment:
Suggest to change to "Publish the image to a public repository or container registry at publication time", as we have a rule that says "do not push directly to a registry", and registry and repository serve two different purposes here.

Reply:
Oh really, where is the point "do not push directly to a registry"? I'll take a look at that and try to revise this, as "push to repository/do not push to registry" could be quite confusing if you're new to Docker.

Reply:
In another rule we suggest to only _use_ images where you can inspect the Dockerfiles. A directly pushed image may contain anything and not be connected to a Dockerfile at all. That recommendation get's more complicated when you factor in security.

Also, in this final rule, I think the focus is on exporting and publication in a data repository, not about a registry where it is easy to use, but a place where it will exists for some years (Docker Hub may be turned of any day, Zenodo or Figshare much more unlikely).

Reply:
Ok, how about "Publicly archive the image used for the publication"?

Reply:
Yes, please go ahead with that change.